src/loaders/svg/tvgSvgLoader.cpp

```diff
@@ -3530,8 +3540,16 @@ bool SvgLoader::header()
             vw = loaderData.doc->node.doc.vw;
             vh = loaderData.doc->node.doc.vh;
 
-            if (!(viewFlag & SvgViewFlag::Width)) w = vw;
-            if (!(viewFlag & SvgViewFlag::Height)) h = vh;
+            if (!(viewFlag & SvgViewFlag::Width)) {
```

Comment:
one more question, 
What do you think if we apply this scaling in _attrParseSvgNode().

Reply:
I think it's good to add flags(or another way) for `sw` and `sh` and do the final calculation in svgSceneBuild() of tvgSvgSceneBuilder.cpp.

Reply:
We are fixing the deferred the calculation messed up.

Reply:
If width or height is defined as `%`, then w,h and sw,sh will be the same value. Therefore, we don't actually need sw or sh separately. So I guess we can define this as a flag.
And the final svg width and height are determined by _applySvgViewFlag(). I think we can calc the final size according to the flag set here.

`w h (0.020000 0.020000), sw sh (0.020000 0.020000)`

```
--- a/src/loaders/svg/tvgSvgLoader.cpp
+++ b/src/loaders/svg/tvgSvgLoader.cpp
@@ -3526,6 +3526,7 @@ bool SvgLoader::header()
         w = 1.0f;
         h = 1.0f;

+printf("w h (%f %f), sw sh (%f %f) \n", loaderData.doc->node.doc.w, loaderData.doc->node.doc.h, loaderData.doc->node.doc.sw, loaderData.doc->node.doc.sw);

```
```
--- a/src/examples/images/favorite_on.svg
+++ b/src/examples/images/favorite_on.svg
@@ -2,7 +2,7 @@
 <!-- Generator: Adobe Illustrator 16.0.3, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
 <!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
 <svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
-        width="80px" height="80px" viewBox="-14 -14 80 80" enable-background="new -14 -14 80 80" xml:space="preserve">
+        width="2%" height="2%" viewBox="-14 -14 80 80" enable-background="new -14 -14 80 80" xml:space="preserve">

```

Reply:
@JSUYA sure, flag would work as well. sorry, I was offline for a few days